router.go

```diff
@@ -51,8 +52,14 @@ type TreeMux struct {
 
 	// The default PanicHandler just returns a 500 code.
 	PanicHandler PanicHandler
+
 	// The default NotFoundHandler is http.NotFound.
 	NotFoundHandler func(w http.ResponseWriter, r *http.Request)
+
+	// The default OptionsHandler is nil. If set, the handler will run on any OPTIONS requests
+	// without an explicit match.
```

Comment:
The current implementation will cause it to run on matched paths that don't have OPTIONS handlers, but it will not match on paths that are completely unmatched. We should make that distinction here. Maybe this: 

> Any OPTIONS request that matches a path without its own OPTIONS handler will use this handler, if set, instead of calling MethodNotAllowedHandler.